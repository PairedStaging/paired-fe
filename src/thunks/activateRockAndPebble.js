import { fetchData } from '../utils';
import { setError, setLoading, setRockAndPebble } from '../actions';
import * as gql from '../queries';

export const activateRockAndPebble = (rockId, pebbleId) => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const query = gql.activateRockPebbleRelationship(rockId, pebbleId);
      const result = await fetchData(query);
      dispatch(setLoading(false));
      dispatch(setRockAndPebble(result.activateRockPebbleRelationship));
    } catch (error) {
      dispatch(setLoading(false));
      dispatch(setError(error.message));
    }
  };
};