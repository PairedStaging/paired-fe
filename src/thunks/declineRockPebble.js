import { fetchData} from '../utils';
import { setError, setLoading, setRockAndPebble } from '../actions';
import * as gql from '../queries';

export const declineRockPebbleRelationship = (rockId, pebbleId, reason) => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const query = gql.declineRockPebbleRelationship(rockId, pebbleId, reason);
      const result = await fetchData(query);
      dispatch(setLoading(false));
      dispatch(setRockAndPebble(result.declineRockPebbleRelationship));
    } catch (error) {
      dispatch(setLoading(false));
      dispatch(setError(error.message));
    }
  };
};