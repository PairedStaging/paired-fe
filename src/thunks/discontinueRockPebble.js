import { fetchData} from '../utils';
import { setError, setLoading, setRockAndPebble } from '../actions';
import * as gql from '../queries';

export const discontinueRockPebbleRelationship = (rockId, pebbleId, reason, userRelationship) => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const query = gql.discontinueRockPebbleRelationship(rockId, pebbleId, reason, userRelationship);
      const result = await fetchData(query);
      dispatch(setLoading(false));
      dispatch(setRockAndPebble(result.discontinueRockPebbleRelationship));
    } catch (error) {
      dispatch(setLoading(false));
      dispatch(setError(error.message));
    }
  };
};