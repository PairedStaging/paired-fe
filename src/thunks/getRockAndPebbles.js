import { fetchData } from '../utils';
import { setError, setLoading, setRockAndPebble } from '../actions';
import * as gql from '../queries';

export const getRockAndPebbles = (id) => {
  return async dispatch => {
    dispatch(setLoading(true));
    try {
      const rocksandpebbleQuery = gql.getUserRockAndPebble(id);
      const rockandpebbleResult = await fetchData(rocksandpebbleQuery);
      dispatch(setRockAndPebble(rockandpebbleResult.getUserRockAndPebble));
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  };
};