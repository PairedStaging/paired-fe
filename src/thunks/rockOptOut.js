import { fetchData } from '../utils';
import { setError, setLoading, setUser } from '../actions';
import * as gql from '../queries';

export const rockOptInOut = (id) => {
  return async dispatch => {
    try {
      dispatch(setLoading(true));
      const query = gql.rockOptInOut(id);
      await fetchData(query);
      const result= gql.getUser(id);
      const updatedUser = await fetchData(result);
      dispatch(setLoading(false));
      dispatch(setUser(updatedUser.getUser));
    } catch (error) {
      dispatch(setLoading(false));
      dispatch(setError(error.message));
    }
  };
};