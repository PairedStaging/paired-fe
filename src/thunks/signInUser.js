import { fetchData } from '../utils';
import { setError, setLoading, setUser } from '../actions';
import { getSchedule } from './getSchedule';
import { updateUser } from './updateUser';
import * as gql from '../queries';
import { getRockAndPebbles } from './getRockAndPebbles';

export const signInUser = (id, photoURL) => {
  return async dispatch => {
    dispatch(setLoading(true));
    try {
      const userQuery = gql.getUserByFirebaseID(id);
      const { user } = await fetchData(userQuery);
      if (user !== null) {
        if (user.image !== photoURL) {
          dispatch(updateUser({ id: user.id, image: photoURL }, true));
        } else {
          dispatch(setUser(user));
        }
        dispatch(getSchedule(user.id));
        dispatch(getRockAndPebbles(user.id));
      } else {
        dispatch(setUser({ isNewUser: true }));
      }
    } catch (error) {
      dispatch(setError(error.message));
    }
    dispatch(setLoading(false));
  };
};
