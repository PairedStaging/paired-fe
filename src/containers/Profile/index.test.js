import React from 'react';
import { shallow } from 'enzyme';
import { Profile, mapDispatchToProps } from './index';
import * as thunk from '../../thunks/createUser';

describe('Profile', () => {
  describe('Profile component', () => {
    let wrapper;
    const mockCreateUser = jest.fn();
    const mockProps = {
      createUser: mockCreateUser,
      email: 'Hill@gmail.com',
      phoneNumber: '1234567890',
      firebaseID: 'bAzN8PSfQSPL9rg4pMhTha8zewn2',
      image: 'https://avatars1.githubusercontent.com/u/49459620?v=4',
      name: 'Hillary',
      match: { patch: '/profile' }
    };

    beforeEach(() => {
      wrapper = shallow(<Profile {...mockProps} />);
      window.fetch = jest.fn();
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should have a default state', () => {
      const expected = {
        name: mockProps.name,
        email: mockProps.email,
        phoneNumber: '',
        phoneNumberError: '',
        program: '',
        module: '',
        pronouns: '',
        slack: '',
        submitted: false,
        'skill 1': '',
        'skill 2': '',
        'skill 3': ''
      };
      expect(wrapper.state()).toEqual(expected);
    });

    it('should return true if the user has not entered info for program, module, email, name, pronouns, or slack', () => {
      const expected = true;
      const result = wrapper.instance().checkDropdowns();
      expect(result).toEqual(expected);
    });

    it('should handleSubmit when the form is submitted', async () => {
      const instance = wrapper.instance();
      jest.spyOn(instance, 'handleSubmit');
      wrapper.instance().forceUpdate();
      wrapper.find('form').simulate('submit', { preventDefault: jest.fn() });
      expect(instance.handleSubmit).toHaveBeenCalled();
    });

    it('should call handleChange when a user types in the input', () => {
      const mockSlack = '@tiffany';
      const mockEvent = {
        target: { name: 'slack', value: mockSlack }
      };
      const initialState = '';
      expect(wrapper.state('slack')).toEqual(initialState);
      wrapper.instance().handleChange(mockEvent);
      expect(wrapper.state('slack')).toEqual(mockSlack);
    });
  });

  describe('mapDispatchToProps', () => {
    window.fetch = jest.fn();
    it('should call dispatch with createUser thunk when createUser is called', () => {
      const mockDispatch = jest.fn();
      thunk.createUser = jest.fn();
      const actionToDispatch = thunk.createUser({}, []);
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.createUser({}, []);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
