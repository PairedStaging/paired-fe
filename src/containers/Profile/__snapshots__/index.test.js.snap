// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Profile component should match the snapshot 1`] = `
<form
  className="Profile--form"
  onSubmit={[Function]}
>
  <h2
    className="Profile--h2"
  >
    Please complete your profile
  </h2>
  <div
    className="Profile--div"
  >
    <div
      className="Profile--div--flex"
    >
      <label
        htmlFor="name"
      >
        Name
        <span
          className="Profile--asterisk"
        >
          *
        </span>
      </label>
      <input
        className="Profile--input"
        name="name"
        onChange={[Function]}
        value="Hillary"
      />
      <label
        htmlFor="Pronouns"
      >
        Pronouns
      </label>
      <input
        className="Profile--input"
        label="Pronouns"
        maxLength="10"
        name="pronouns"
        onChange={[Function]}
        placeholder="e.g. she/her, he/him, they/them, etc."
        value=""
      />
      <label
        htmlFor="email"
      >
        Email
        <span
          className="Profile--asterisk"
        >
          *
        </span>
      </label>
      <input
        className="Profile--input"
        name="email"
        onChange={[Function]}
        value="Hill@gmail.com"
      />
      <label
        htmlFor="phoneNumber"
      >
        Phone Number
      </label>
      <div
        id="phone-number-error"
      />
      <input
        className="Profile--input"
        name="phoneNumber"
        onChange={[Function]}
        placeholder="e.g. 3031234567"
        value=""
      />
      <label
        htmlFor="slack"
      >
        Slack handle
        <span
          className="Profile--asterisk"
        >
          *
        </span>
      </label>
      <input
        className="Profile--input Profile--input-slack"
        name="slack"
        onChange={[Function]}
        placeholder="@"
        value=""
      />
    </div>
    <Dropdown
      className="Profile--Dropdown"
      handleChange={[Function]}
      label="Program"
      options={
        Array [
          "FE",
          "BE",
        ]
      }
      required={true}
      selectedItem=""
    />
    <Dropdown
      className="Profile--Dropdown"
      handleChange={[Function]}
      label="Module"
      options={
        Array [
          1,
          2,
          3,
          4,
          "Graduate",
        ]
      }
      required={true}
      selectedItem=""
    />
    <h4>
      Provide skills you can help others with
    </h4>
    <Dropdown
      className="Profile--Dropdown"
      handleChange={[Function]}
      label="Skill 1"
      options={
        Array [
          "grid",
          "flexbox",
          "mythical creatures",
          "react",
          "jsFun",
          "rspec",
          "rails",
          "ruby",
          "jest/enzyme",
          "docker",
          "lightning talks",
          "javascript",
          "css",
          "active record",
          "SQL",
          "sinatra",
        ]
      }
      selectedItem=""
    />
    <Dropdown
      className="Profile--Dropdown"
      handleChange={[Function]}
      label="Skill 2"
      options={
        Array [
          "grid",
          "flexbox",
          "mythical creatures",
          "react",
          "jsFun",
          "rspec",
          "rails",
          "ruby",
          "jest/enzyme",
          "docker",
          "lightning talks",
          "javascript",
          "css",
          "active record",
          "SQL",
          "sinatra",
        ]
      }
      selectedItem=""
    />
    <div
      className="Profile--div--flex"
    >
      <label
        htmlFor="Skill 3"
      >
        Skill 3
      </label>
      <input
        className="Profile--input"
        maxLength="20"
        name="Skill 3"
        onChange={[Function]}
        placeholder="Additional skill - max 20 characters"
        value=""
      />
    </div>
  </div>
  <p
    className="Profile--message"
  />
  <button
    className="Profile--button"
    disabled={true}
  >
    Submit
  </button>
</form>
`;
